### DISCUSSION TITLE
Make Delete button appear only for User who created Message in Channel

### DISCUSSION THREAD

**Discussion by BenJarrett** [nss-evening-cohort-14/evening-client-side#237] [Help Me!] [answered]
- Ben Jarrett
 
 - I cannot find a ticket that talks about this at all. 

 - In our Slack Clone project, I am having trouble writing the logic to only show the delete button on only the messages that the signed-in user created. 
 - We only want the user who created the message in the channel to be able to delete the message that they have created.

- There are no errors, just need to be pointed in the right direction to write this logic. 

 - https://www.digitalocean.com/community/tutorials/7-ways-to-implement-conditional-rendering-in-react-applications
 - https://stackoverflow.com/questions/62041553/hide-or-show-a-button-for-authenticated-user-in-react
 - https://flexiple.com/react/conditional-rendering-in-react/

- I've tried using a switch statement. 
- I've tried passing the user, uid, and messageID props and adding logic to only show if that matches the user

 - https://github.com/BenJarrett/slack-clone/tree/bj-delete-message-functionality

**Comment by drteresavasquez** [comment]
We have done this a couple of times. Check to see if the user id matches the id of the person who created the message and either show the button or don’t if they don’t. Look at the repo where we did the logic for if the button should show or not.

**Comment by drteresavasquez** [comment]
This example shows text being shown based on logic but it’s the same idea. https://github.com/nss-evening-cohort-14/all-things-react/blob/main/src/components/StudentCard.js#L41

**Reply by BenJarrett** [reply]
Got it. I'll look through this and figure it out. 
I will let you know if I have any questions, thank you!

**Reply by BenJarrett** [reply]
Does anything change if the onClick is being called on a button that is being mapped as the messages are being dynamically created?

**Reply by drteresavasquez** [reply]
You mean if the button is not rendered due to logic? If so, it should be ok. Do you have an example you want to show? I can jump on in a few minutes.

**Reply by BenJarrett** [reply]
Hey Dr. T. Thanks for all the help. Trinity came in and helped through this bc  I was confused if I needed to use a switch statement or just add a ternary on the button mapping. 
I'll add the answer below!

**Comment by BenJarrett** [comment, marked as answer]
So because this button is being added to the message map, It was displaying next to the message that was being dynamically created when mapped. 

So instead of useing a switch statement, I added a ternary before the button with the handleDeleteMessageButton functionality. 

``` <div>
            { msgObj.userUID === user.uid ? <Button
          style={{
            backgroundColor: 'transparent',
            borderWidth: '0rem'
          } }
          onClick={() => (handleDeleteMessageButton(msgObj.messageID))}>
          <svg
                    xmlns="http://www.w3.org/2000/svg"
                    width="16"
                    height="16"
                    fill="black"
                    className="bi bi-trash"
                    viewBox="0 0 16 16">
                   <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="black" viewBox="0 0 16 16">
                    <path d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"/>
                    <path fillRule="evenodd" d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"/>
                    </svg>
                  </svg>
          </Button> : '' }
        </div>

        
In this ternary, I needed to say "if the userUID of the message is equal to the current user, show the delete button, if not, show an empty string".
In order to do this, I needed to access the user. 
In the MessagesView, I passed MessageBubble the user prop. 
In the MessageBubble component, I passed user as a prop.
